tests/phpunit/tests/db.php

```diff
@@ -1771,6 +1761,36 @@ public function data_prepare_with_placeholders() {
 				false,
 				"WHERE `evil_{$wpdb->placeholder_escape()}s_field` = 321;",
 			),
+			array(
+				'WHERE %i = %d;',
+				array( 'value`', 321 ),
+				false,
+				'WHERE `value``` = 321;',
+			),
+			array(
+				'WHERE `%i = %d;',
+				array( ' AND evil_value', 321 ),
+				false,
+				'WHERE `` AND evil_value` = 321;', // Won't run (SQL parse error: "Unclosed quote").
+			),
+			array(
+				'WHERE %i` = %d;',
+				array( 'evil_value -- ', 321 ),
+				false,
+				'WHERE `evil_value -- `` = 321;', // Won't run (SQL parse error: "Unclosed quote").
+			),
+			array(
+				'WHERE `%i`` = %d;',
+				array( ' AND true -- ', 321 ),
+				false,
+				'WHERE `` AND true -- ``` = 321;', // Won't run (Unknown column '').
+			),
+			array(
+				'WHERE ``%i` = %d;',
+				array( ' AND true -- ', 321 ),
+				false,
+				'WHERE ``` AND true -- `` = 321;', // Won't run (SQL parse error: "Unclosed quote").
+			),
```

Comment:
An additional test to make sure the ordered placeholders work. 

```suggestion
			array(
				'WHERE ``%i` = %d;',
				array( ' AND true -- ', 321 ),
				false,
				'WHERE ``` AND true -- `` = 321;', // Won't run (SQL parse error: "Unclosed quote").
			),
			array(
				'WHERE ``%2$i` = %1$d;',
				array( '1', 'two' ),
				false,
				'WHERE `two` = 1;',
			),
```

Reply:
Good point, incase ordered placeholders are handled differently for identifiers in the future... I did tweak your test a bit, so the `%2$i` is not quoted (to match how it would be used).

Reply:
This was added in commit [142200e](https://github.com/WordPress/wordpress-develop/pull/2192/commits/142200e19d94e4c73b15b3c686ad55d57e837ca6).